defmt.x.in

```diff
@@ -11,36 +11,8 @@ SECTIONS
     /* Format implementations for primitives like u8 */
     *(.defmt.prim.*);
 
-    /* ERROR logging level */
-    _defmt_error_start = .;
-    *(.defmt.error.*);
-    _defmt_error_end = .;
-
-    /* WARN logging level */
-    _defmt_warn_start = .;
-    *(.defmt.warn.*);
-    _defmt_warn_end = .;
-
-    /* INFO logging level */
-    _defmt_info_start = .;
-    *(.defmt.info.*);
-    _defmt_info_end = .;
-
-    /* DEBUG logging level */
-    _defmt_debug_start = .;
-    *(.defmt.debug.*);
-    _defmt_debug_end = .;
-
-    /* TRACE logging level */
-    _defmt_trace_start = .;
-    *(.defmt.trace.*);
-    _defmt_trace_end = .;
-
-    /* Format/write! strings */
-    *(.defmt.fmt.*);
-
-    /* User interned strings (Str) */
-    *(.defmt.str.*);
+    /* Everything user-defined */
+    *(.defmt.sym.*);
```

Comment:
Then I'd have to use `*(.defmt.*);` instead, which seems like it'd place all symbols in `.defmt.prim` twice

Reply:
Mixing `.defmt.prim` and `*(.defmt.*)` should work fine: it shouldn't cause "duplicate symbols" errors (I hope)

Reply:
That appears to work, although I don't understand why. Does the linker only place the `.defmt.prim` input sections once, even if later directives match them again?

Note that we still append a generated section name after the `.defmt` (which, I guess, we have to, if we want to benefit from link-time garbage collection).

Reply:
> Does the linker only place the .defmt.prim input sections once, even if later directives match them again?

that matches what I have seen so yeah, I think so

> Note that we still append a generated section name after the .defmt (which, I guess, we have to, if we want to benefit from link-time garbage collection).

Yes, we need each symbol to be in its own separate input section because the linker drops unused *sections*, not individual symbols.